src/utils/framework.ts

```diff
@@ -399,3 +416,67 @@ export const getMatrix2dToc = (framework: FrameworkFields | undefined): MatrixTo
 
     return toc.filter(isDefined);
 };
+
+export function getScaleWidgetsData(framework: FrameworkFields, entry: Entry) {
+    if (!framework || !framework.widgets) {
+        return [];
+    }
+
+    const { widgets } = framework;
+
+    const scaleWidgets = widgets
+        .filter(w => w.widgetId === 'scaleWidget')
```

Comment:
We should use widgetId on widgets to identify if it is ScaleWidget, we wil no longer need to cast it ourself

Reply:
@tnagorra what do you mean?

Reply:
What the filter does is act as a type guard, so if that widget has a string literal as identifier for different unions we wont' need to cast explicitly.